### DISCUSSION TITLE
custom across_sth function

### DISCUSSION THREAD

**Discussion by bnicenboim** [markfairbanks/tidytable#342] [Q&A] [answered]
Hi!

I'm looking at utils-prep_exprs.R, and I was wondering how can I create a custom `across_sth()` function for my own package. 

My use case is a bit more complicated than this, but say that I want to allow the user to use `across_num(...)` and `across_x(...)` where
`across_num(...)` gets transformed in `across.(where(is.numeric), ...)` and `across_x(...)` gets transformed in `across.(dplyr::starts_with("x), ...)`.
I cannot figure a way to hack your code to do this. Any hints?

Best,
Bruno

**Comment by markfairbanks** [comment, marked as answer]
Basic steps:
1. Catch that the user is calling `across_num()`
2. Extract the arguments the user passed to `across_num()`
3. Place those arguments inside of a call to `across.()` with `where(is.numeric)`
4. Send this new `across.()` call back through `prep_expr()`

So you'll need to add this `else if` to `prep_expr()` (and don't forget to add `"across_num"` to the `call_fns` vector!):

```r
} else if (is_call(x, "across_num")) {
  call_dots <- as.list(x[-1])
  x <- call2("across.", expr(where(is.numeric)), !!!call_dots)
  prep_expr(x, data, {{ .by }}, j = j)
}
```

Hope this helps - if you have any questions or need anything clarified let me know.

**Reply by bnicenboim** [reply]
thanks for the fast answer!

**Reply by bnicenboim** [reply]
ok, this made me found a related problem in #345 :/
